rfd/0079/project-service-draft.md

```diff
@@ -0,0 +1,121 @@
+
+All below endpoints may be prefixed with `/users/$userId` to access data for a user other than the one you're authenticated as, assuming authorization is granted. For example:
+
+`GET /users/abc123/projects` will return the same data as a request to `/projects` would if you were authenticated as `abc123`. 
+
+### /projects
+
+Endpoint operations:
+
+* GET - A list of projects associated with requesting user
+* POST - Create a new project via YAML manifest transmitted in POST payload
+
+### /projects/$projectId
+
+Endpoint operations:
+
+* GET - Project data and metadata derived from manifest and meta file in JSON format for specified project ID
+* PUT - Update project data and/or metadata via JSON document transmitted in PUT payload
+* DELETE - Mark project as removed
+
+### /projects/$projectId/manifest
+
+Endpoint operations:
+
+* GET - Get project manifest in YAML format as it was provided
+* PUT - Update project manifest by sending YAML payload
+
+### /projects/$projectId/metadata
+
+Endpoint operations:
+
+* GET - Get project metadata in YAML format as it was provided
+* PUT - Update project metadata by sending YAML payload
+
+### /projects/$projectId/metadata/$key
+
+Endpoint operations:
+
+* GET - Returns value associated with metadata key
+* PUT - Update value for metadata key
+* DELETE - Remove metadata key
+
+### /projects/$projectId/tags
+
+Endpoint operations:
+
+* GET - list of tags associated with project, automatic and custom
+* POST - Create a user-defined tag, associated with a specified sha1sum tag
+
+### /projects/$projectId/tags/$tagId
+
+Endpoint operations
+
+* GET - Detailed project information and manifest for the version associated with the given tag ID
+* DELETE - Remove a user-specified tag, this will return an error if user attempts to delete a sha1 sum tag
+
+### /projects/$projctId/services
+
+Endpoint operations:
+
+* GET - Summarized service information for specified project
+* POST - Create a new service for the specified project via service manifest transmitted in POST payload
+
+### /projects/$projectId/state
+
+Endpoint operations:
+
+* GET - Goal/Actual state information via a proxied request to the Convergence service
+
+### /services
+
+Endpoint operations:
+
+* GET - A list of all user services across all projects
+* POST - Create a new service via service manifest transmitted in POST payload; the associated project must be specified in the manifest
+
+### /services/$serviceId
+
+Alias: `/projects/$projectId/services/$serviceId`
+
+Endpoint operations:
+
+* GET - Detailed service information and manifest for specified service ID
+* PUT - Updated service manifest via new manifest transmitted in PUT payload
+* DELETE - Mark service as removed
```

Comment:
Yep. I picked up on this because this method might suggest a data structure that persists the service in the manifest, but adds some metadata indicating its deleted.

The service might persist in older versions of the manifest, but anybody inspecting the current project manifest after hitting this should not expect to see any sign of the deleted service.